### ISSUE TITLE
Inspect Object not working

### ISSUE THREAD

**Issue by janodvarko** [firebug/firebug#8077]
This has been originally reported here:
https://bugzilla.mozilla.org/show_bug.cgi?id=1335853

What did you do?
================
Hover over an object on a page, right-click-->Inspect Element with Firebug

What happened?
==============
This toggles the inspector window on or off and when it toggles it on, it does not go to the html object - instead it's at the default body tag.

What should have happened?
==========================
It should not toggle the inspector window -- it should turn it on if it was off, but not toggle it off. It should also drill down to the object within the html.

Is there anything else we should know?
======================================
FireBug worked correctly. But you probably hear this a lot.

---

Looking at the Firebug code, the stack that leads to opening devtools is:
- BrowserOverlay.js::startFirebug
- BrowserOverlay.js::showMultiprocessNotification
- BrowserOverlay.js::toggleDevTools
- BrowserOverlay.js::showToolbox

And in the end we simply call DevTools.gDevTools.showToolbox(target);

Honza

**Comment by captainbrosset** [comment]
@janodvarko and I discussed about this. It is very confusing for people who are not using Firebug anymore to see 2 "inspect element" menu items in the list, and have one that doesn't work correctly.

We should keep the menu for people who have e10s off, because in that case they still use Firebug.
But for everyone else who don't have a choice, and are now using the built-in devtools, it seems just simpler to just hide the menu altogether.

I did some investigation, the menu item is added here:
https://github.com/firebug/firebug/blob/master/extension/content/firebug/firefox/browserMenu.js#L439-L447

And it is possible to check for multi-processes (e10s) with the `BrowserOverlay.isMultiprocessEnabled` function. A usage example is here:
https://github.com/firebug/firebug/blob/master/extension/content/firebug/firefox/browserMenu.js#L105

**Comment by SebastianZ** [comment]
Created https://github.com/firebug/firebug/pull/8079 for this.

Sebastian

**Comment by captainbrosset** [comment]
Oops, I didn't see @SebastianZ had jumped on this :)

**Comment by SebastianZ** [comment]
No problem. Thank you for working on this! As far as I can see, the only difference in our implementations is that I am additionally checking for Aurora, which actually may not be necessary.

@janodvarko Can you please have a look at the PRs?

Sebastian

**Comment by janodvarko** [comment]
> @janodvarko Can you please have a look at the PRs?

I already did, see  my review comment.

Honza

**Comment by captainbrosset** [comment]
Thanks @janodvarko and @SebastianZ ! What's the process now to get this landed and out to users?

**Comment by janodvarko** [comment]
> Thanks @janodvarko and @SebastianZ ! What's the process now to get this landed and out to users?

As soon as the patch is reviewed and landed, I'll build new xpi (version updated) and upload to AMO for review. The review can take couple of days and then it's available to all users.

Honza

**Comment by janodvarko** [comment]
New version (2.0.19) is now waiting for review on AMO.

Honza

**Comment by SebastianZ** [comment]
@janodvarko thank you for pushing the new version online! Can you please also
1. add a note to the AMO site telling people that Firebug is discontinued and people get automatically switched to the DevTools and link to the migration guide?
2. update https://getfirebug.com/downloads/ to the new version?

@captainbrosset I don't want to discourage you, but the comments on AMO about "the new Firebug version" are pretty bad. The DevTools urgently need some UX improvements. I'll try to help if I can find some spare time.

Sebastian

**Comment by janodvarko** [comment]
add #1) AMO updated.
add #2) I am doing this as soon as the new version passes AMO review.

Honza

**Comment by janodvarko** [comment]
> the comments on AMO about "the new Firebug version" are pretty bad. The DevTools
> urgently need some UX improvements. 

Agree on this.

I am currently focusing on the Netmonitor panel (as part of Netmonitor.html project, i.e. removing XUL and Chrome API). My plan for next steps is triaging all `firebug-gaps` reports and collect UI/UX suggestions. As soon as the Net panel refactoring phase is done we can start improve the UI.

Later, we might want to do similar things for other panels.

Honza

**Comment by SebastianZ** [comment]
> My plan for next steps is triaging all `firebug-gaps` reports and collect UI/UX suggestions. As soon as the Net panel refactoring phase is done we can start improve the UI.

For reference, I listed the (in my eyes) most pressing Firebug gaps in [bug 991806 comment 20](https://bugzilla.mozilla.org/show_bug.cgi?id=991806#c20). As there were rather few comments until then, it's rather a guess, though, and may not be up-to-date. E.g. several people were arguing that the DevTools are missing FirePath features.
Also, this list doesn't contain UI/UX issues unrelated to Firebug, e.g. [bug 1324254](https://bugzilla.mozilla.org/show_bug.cgi?id=1324254), which was mentioned several times (and is somewhat related to this issue here).

Sebastian

**Comment by inspector71** [comment]
I resent being pushed an update like this. It's arguably one step short of arbitrarily, remotely disabling functionality in people's software. 

FWIW I've been trying to switch over to devtools since it's being forced on to all developers who want to remain loyal to Firefox. It's just inferior to Firebug - STILL - after **years** of development (and the corresponding lack of development in Firebug), The Firebug theme for Devtools isn't very accurate at all.

Is it realistic for a front-end web dev with minimal Firefox ('chrome'-side) development experience to contribute to DevTools through fixing the inconsistencies in the Firebug DevTools theme? I guess I"ll see if I can try starting with Stylish.

**Comment by SebastianZ** [comment]
> Is it realistic for a front-end web dev with minimal Firefox ('chrome'-side) development experience to contribute to DevTools through fixing the inconsistencies in the Firebug DevTools theme?

Yes, most parts of the DevTools UI are pure HTML, CSS and JavaScript. You should read how to get involved at https://wiki.mozilla.org/DevTools/GetInvolved, though.

Sebastian
